From 8dd04447e51d9ad1c8419bd0f3feab153544019c Mon Sep 17 00:00:00 2001
From: "i.MX Yocto Project Build" <lauren.post@nxp.com>
Date: Fri, 4 May 2018 04:14:50 -0500
Subject: [PATCH] Remove ion related code in kernel 4.14

- Remove ion related code in kernel 4.14
as ion is not ready yet

Signed-off-by: Lyon Wang<lyon.wang@nxp.com>
---
 libs/gstimxcommon.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libs/gstimxcommon.c b/libs/gstimxcommon.c
index e1a81c7..0e0aa9f 100644
--- a/libs/gstimxcommon.c
+++ b/libs/gstimxcommon.c
@@ -23,6 +23,7 @@
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <linux/version.h> 
 #ifdef USE_ION
 #include <linux/ion.h>
 #endif
@@ -41,6 +42,7 @@ unsigned long phy_addr_from_fd(int dmafd)
     return NULL;
   }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 1, 4) 
   struct ion_phys_dma_data data = {
     .phys = 0,
     .size = 0,
@@ -61,6 +63,9 @@ unsigned long phy_addr_from_fd(int dmafd)
 #else
   return NULL;
 #endif
+#else
+  return NULL;
+#endif
 }
 
 unsigned long phy_addr_from_vaddr(void *vaddr, int size)
@@ -76,6 +81,7 @@ unsigned long phy_addr_from_vaddr(void *vaddr, int size)
     return NULL;
   }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 1, 4) 
   struct ion_phys_virt_data data = {
     .virt = (unsigned long)vaddr,
     .phys = 0,
@@ -96,4 +102,7 @@ unsigned long phy_addr_from_vaddr(void *vaddr, int size)
 #else
   return NULL;
 #endif
+#else
+  return NULL;
+#endif
 }
-- 
2.7.4

